app/src/main/java/com/safetyheads/akademiaandroida/YouTube/repository/VideoRepository.kt

```diff
@@ -0,0 +1,10 @@
+package com.safetyheads.akademiaandroida.YouTube.repository
+
+import com.safetyheads.akademiaandroida.YouTube.entities.video.YouTubeVideoDataClass
+import com.safetyheads.akademiaandroida.network.NetworkResult
+import kotlinx.coroutines.flow.Flow
+
+interface VideoRepository {
+
+    suspend fun getYouTubeVideo(previousFilmDate: String): Flow<NetworkResult<YouTubeVideoDataClass>>
```

Comment:
`previousFilmDate` Hmm why you use film date to get video from YT API?

Reply:
There is no better way to download a specific video from YT Api. The 'previousFilmDate' method gets the movie and creates the date before the downloaded movie, i.e. (downloaded movie: 2023.04.17 22:00:00), then I set 'previousFilmDate' (2023.04.16 22:00:00) and download the movie before this date to download several videos in sequence. To minimize API mistakes, I subtract one day (the intensity of videos on the channel is not high), because the YT API works quite poorly in this direction.